code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -30,14 +36,25 @@
 	if(!friend.client && friend_initialized)
 		addtimer(CALLBACK(src, PROC_REF(reroll_friend)), 1 MINUTES)
 
+	if(HAS_TRAIT(owner, TRAIT_PERCEPTUAL_TRAUMA_BYPASS))
+		QDEL_NULL(friend)
+
 /datum/brain_trauma/special/imaginary_friend/on_death()
 	..()
 	qdel(src) //friend goes down with the ship
 
 /datum/brain_trauma/special/imaginary_friend/on_lose()
 	..()
+	UnregisterSignal(owner, list(SIGNAL_REMOVETRAIT(TRAIT_PERCEPTUAL_TRAUMA_BYPASS), SIGNAL_ADDTRAIT(TRAIT_PERCEPTUAL_TRAUMA_BYPASS)))
 	QDEL_NULL(friend)
 
+/datum/brain_trauma/special/imaginary_friend/proc/temporary_bypass()
+	if(keep_without_friend) // if two trait sources
```

Comment:
Signal handler

Reply:
fuck imaginry friaends. the power of firensdhip wins over anomalies